tests/monotouch-test/System.Net.Http/MessageHandlers.cs

```diff
@@ -758,5 +758,61 @@ public void GHIssue8344 ()
 				Assert.AreEqual (HttpStatusCode.Unauthorized, httpStatus, "Second status not ok");
 			}
 		}
+		class TestDelegateHandler : DelegatingHandler {
+			public bool FirstRequestCompleted { get; private set; } = false;
+			public bool SecondRequestCompleted { get; private set; } = false;
+			protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+			{
+				// test that we can perform a retry with the same request
+				var _ = await base.SendAsync(request, cancellationToken);
+				FirstRequestCompleted = true;
```

Comment:
You're not verifying that what was sent was correct. I believe the actual request text can be found in the response from httpbin.org.

Reply:
@mandel-macaque could you please help with this?